__tests__/pages/_app.test.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import {
+  render,
+  waitFor,
+  waitForElementToBeRemoved
+} from '@testing-library/react'
+import { MockedProvider } from '@apollo/client/testing'
+import MyApp from '../../pages/_app'
+import Login from '../../pages/login'
+import * as Sentry from '@sentry/browser'
+import GET_APP from '../../graphql/queries/getApp'
+import dummyLessonData from '../../__dummy__/lessonData'
+import dummySessionData from '../../__dummy__/sessionData'
+import dummyAlertData from '../../__dummy__/alertData'
+import posthog from 'posthog-js'
+
+jest.mock('posthog-js')
+jest.spyOn(Sentry, 'captureException')
+
+const mocks = [
+  {
+    request: { query: GET_APP },
+    result: {
+      data: {
+        session: dummySessionData,
+        lessons: dummyLessonData,
+        alerts: dummyAlertData
+      }
+    }
+  }
+]
+
+describe('MyApp component', () => {
+  const OLD_ENV = process.env
+  beforeEach(() => {
+    jest.clearAllMocks()
+    process.env = { ...OLD_ENV }
+  })
+  afterEach(() => {
+    process.env = OLD_ENV
+  })
+
+  test('posthog init function should not be called if not in production environment', async () => {
+    const { queryByText } = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <MyApp Component={Login} />
+      </MockedProvider>
+    )
+    await waitForElementToBeRemoved(() => queryByText('Loading...'))
+    await (() => expect(posthog.init).not.toHaveBeenCalled())
+  })
+
+  test('posthog init function is being called with the correct arguments', async () => {
+    process.env = {
+      ...process.env,
+      NODE_ENV: 'production',
+      POSTHOG_API_KEY: 'fake-posthog-api-key'
+    }
+    const { queryByText } = render(
+      <MockedProvider mocks={mocks} addTypename={false}>
+        <MyApp Component={Login} />
+      </MockedProvider>
+    )
+    await waitForElementToBeRemoved(() => queryByText('Loading...'))
+    await (() =>
+      expect(posthog.init).toHaveBeenCalledWith(process.env.POSTHOG_API_KEY, {
+        api_host: 'https://app.posthog.com'
+      }))
+    await (() => console.log(process.env))
```

Comment:
Why are we logging this value?

Reply:
Also you could remove the `await` on the `expect` call on both tests, it's not doing anything there.

If the tests fails then do `await waitFor(() => expect(...))`

Reply:
Here's an example of when to use `waitFor`:
Let's say we are making a test for the login page such that when someone presses the submit button with incorrect input fields, a popup appears saying `Lol u needa put correct values in the fields bro`. In order to test if the popup appeared with that test, you might want to do something like 
```
await waitFor(() => getByText('Lol u needa put correct values in the fields bro'))
```
This is because that popup might take some time to render, so you might need to wait with `waitFor` until it finishes rendering